doc/tutorials/validation.md

```diff
@@ -133,41 +133,110 @@ kdb setmeta spec/tutorial/spec/test check/validation/match LINE
+of validation and finding the files that hold your configuration from the users.
+At the moment a user still has to know which files should hold the configuration
+and which plugins must be loaded when he mounts configuration files.
+
+This problem can be addressed by recognizing that the location of the
+configuration files and the plugins that must be loaded is part of the
+_schema_ of our configuration and therefore should be stored in the spec namespace.
+
+### Specfiles ###
+We call the files, that contain a complete schema for configuration
+below a specific path in form of metadata, _Specfiles_.
+
+Particularly a _Specfile_ contains metadata that defines
+- the mountpoints of paths,
+- the plugins to load and
+- the behavior of these plugins.
+
+Let us create an example _Specfile_ in the [ni format](/doc/src/plugins/ni/README.md),
+which supports metadata:
+```sh
+kdb mount tutorial.ini spec/tutorial ni
+cat << HERE | kdb import spec/tutorial ni
+[]
+ mountpoint = tutorial.dump
+ infos/plugins = dump validation
+
+[/links/_]
+check/validation = https?://.*\..*
+check/validation/match = LINE
+check/validation/message = not a valid URL
+description = A link to some website
+HERE
+kdb lsmeta spec/tutorial
+#> infos/plugins
+#> mountpoint
+```
+We now have all the metadata that we need to mount and validate the data below
+`/tutorial` in one file.
 
-As next step we need to correctly mount all validation plugins to `/tutorial/spec`.
-Because we have an extra specification of the validation, we can use this
-specification to assemble the plugins. We only have to specify where the
-mountpoint is (and which file name should be used), then we can mount
-the file according the specification and have validation for all namespaces:
+Now we apply this _Specfile_ to the key database to all keys below `tutorial`.
+```sh
+kdb spec-mount /tutorial
+```
+This command automatically mounts `/tutorial` to the backend `tutorial.dump` and
+loads the validation plugin.
 
 ```sh
-kdb setmeta spec/tutorial/spec mountpoint spec-tutorial.dump
-kdb spec-mount /tutorial/spec
-kdb set /tutorial/spec/test wrong
-#> Using name user/tutorial/spec/test
+kdb set /tutorial/links/url "invalid url"
+#> Using name user/tutorial/links/elektra
 # STDERR-REGEX: .*Sorry, the error .#42. occurred ;(⏎
 #               Description: Key Value failed to validate⏎
-#               .*Reason: Not a number.*
+#               .*Reason: not a valid URL.*
 # RET:          5
 ```
 
+Note that the backend `tutorial.dump` is mounted for all namespaces:
+```sh
+kdb file user/tutorial
+# STDOUT-REGEX: /home/.*/\.config/tutorial\.dump
+kdb file system/tutorial
+#> /etc/kdb/tutorial.dump
+kdb file dir/tutorial
+# STDOUT-REGEX: /.*/\.dir/tutorial\.dump
+```
+
+If you want to set a key for another namespace and do not want to go without validation,
+consider that the spec plugin works only when you use cascading keys.
+You can work around that by setting the keys with the `-N` option:
```

Comment:
It is the other way round. Using `system/tutorial/links/elektra` is a workaround to disable validation. The goal is to always use cascading keys.